### PR TITLE
Implement SHL opcode

### LINKED ISSUE #26: Opcode: SHL

### PR DESCRIPTION
**Pull request by FrancoGiachetta**
Closes #26 
This PR handles the SHL opcode

### CODE REVIEW

File: src/codegen/operations.rs [hunk 1]
```diff
@@ -679,6 +683,86 @@ fn codegen_addmod<'c, 'r>(
...
+    let result = ok_ok_block
+        .append_operation(arith::shli(value, shift, location))
+        .result(0)?
+        .into();
+
+    stack_push(context, &ok_ok_block, result)?;
+
+    ok_ok_block.append_operation(cf::br(&empty_block, &[], location));
+
+    // if shifht is grater than 255
```
Comment (MegaRedHand) [inline src/codegen/operations.rs:749]: typo
```suggestion
    // if shift is greater than 255
```

**Review by MegaRedHand — CHANGES_REQUESTED** [review]
Looks good, but please add the gas-related changes

File: src/codegen/operations.rs (hunk 1 above)
Comment (FrancoGiachetta) [reply src/codegen/operations.rs:749]: Oh, my bad. Checking gas addition.

**Review by Oppen — APPROVED** [review]

**Review by MegaRedHand — APPROVED** [review]
LGTM

**Pull request merged**